kaska/kaska.py

```diff
@@ -110,54 +115,60 @@ def _run_smoother(self, dates, parameter_block):
         lai = -2 * np.log(parameter_block[-2, :, :, :])
         cab = -100*np.log(parameter_block[1, :, :, :])
         cbrown = parameter_block[2, :, :, :]
+        if self.save_sgl_inversion is True:
+            save_output_parameters(dates, self.observations, 
+                self.output_folder/"single_imgs/", ["lai", "cab", "cbrown"],
+                           [lai, cab, cbrown], output_format="GTiff",
+                           chunk=self.chunk, fname_pattern="s2_sgl",
+                           options=['COMPRESS=DEFLATE',
+                                    'BIGTIFF=YES',
+                                    'PREDICTOR=1',
+                                    'TILED=YES'])
+               
         # Basically, remove weird values outside of boundaries, nans and stuff
         # Could be done simply with the previously stated data structure, as
         # this is a bit of an adhoc piece of code.
-        lai[~np.isfinite(lai)] = 0
-        cab[~np.isfinite(cab)] = 0
-        cbrown[~np.isfinite(cbrown)] = 0
-        lai[~(lai > 0)] = 0
-        cab[~(cab > 0)] = 0
-        cbrown[~(cbrown > 0)] = 0
+        lai[~np.isfinite(lai)] = np.nan
+        cab[~np.isfinite(cab)] = np.nan
+        cbrown[~np.isfinite(cbrown)] = np.nan
+        lai[lai < 0] = np.nan
```

Comment:
This is not the same, as before. 
What happens when values are `0`? Before they were kept as 0, as the rest of cases. Now, they are kept as 0, but the others converted to `NaN`.

Reply:
The very next use of these arrays is as an argument to interp1d. The docstring of interp1d says:

> calling `interp1d` with NaNs present in input values results in undefined behaviour.

So I feel these really shouldn't be being assigned NaN. I will ask @jgomezdans when I next see him.

Reply:
Oh, it's not that interp1d

Reply:
fix_interp provides a version of scipy's interp1d that deals with NaNs in the array gracefully (and fast). So we use NaNs to indicate missing data. 0 is a legitimate value, but values <0 are not sensible, so we fix them to 0 (or NaN, can't quite remember)